lib/src/player.dart

```diff
@@ -25,41 +25,37 @@ class Player with PlayerEventHandler implements PlayerApi {
   Player({
     this.config = const PlayerConfig(),
   }) {
-    _initializationResult = _completer.future;
     _uuid = hashCode.toString();
+    _nativePlayerHandle = _createNativePlayer();
+  }
 
+  Future<_NativePlayerHandle> _createNativePlayer() async {
     final mainChannel = ChannelManager.registerMethodChannel(
       name: Channels.main,
     );
 
-    mainChannel
-        .invokeMethod<bool>(
-      Methods.createPlayer,
-      Map<String, dynamic>.from(
-        {
-          'id': id,
-          'playerConfig': config.toJson(),
-        },
-      ),
-    )
-        .then((value) {
-      if (value == null || value == false) {
-        _completer.complete(false);
-        return;
+    final success = await mainChannel.invokeMethod<bool>(
+        Methods.createPlayer,
+        Map<String, dynamic>.from(
+          {
+            'id': id,
+            'playerConfig': config.toJson(),
+          },
+        ),
+      );
+      if (success == null || success == false) {
+        throw Exception('Error initializing player on native platform side.');
       }
-
-      _methodChannel = ChannelManager.registerMethodChannel(
+      final methodChannel = ChannelManager.registerMethodChannel(
         name: '${Channels.player}-$id',
         handler: _playerMethodCallHandler,
       );
-
-      _eventChannel = ChannelManager.registerEventChannel(
+      final eventChannel = ChannelManager.registerEventChannel(
         name: '${Channels.playerEvent}-$id',
+        onEvent: onPlatformEvent,
       );
-      _eventChannel.receiveBroadcastStream().listen(onPlatformEvent);
 
-      _completer.complete(true);
```

Comment:
Completer is used to bridge a callback based API to Future. Given that we are chaining Futures, it is not needed.